client/src/assets/player/peertube-player.ts

```diff
@@ -29,10 +29,15 @@ function getVideojsOptions (options: {
   peertubeLink: boolean,
   poster: string,
   startTime: number
-  theaterMode: boolean
+  theaterMode: boolean,
+  controls?: boolean,
+  muted?: boolean,
+  loop?: boolean
 }) {
   const videojsOptions = {
-    controls: true,
+    controls: options.controls !== undefined ? options.controls : true,
```

Comment:
This can be manage via ES6 parameters default value.

ie : `controls: boolean = true,`

Reply:
I'm not sure what you mean. This is explicitly defaulting `controls` to be `true` only when `options.controls` is exactly undefined. It is not the same to just set it to true here. Also the syntax you provide is for defining classes but this is a raw object.

Reply:
Sorry, it could be written like that : `controls = true,`.

For me, that's why default parameters values is for, if controls is undefined or not specified (so undefined), it will be set to true. Otherwise, it'll take whatever value you pass. It's just syntactic sugar to do the same thing.

Reply:
No, if I wrote `controls = true` it would be permanently set to `true` regardless of the intention of the caller to this function, (as well as give me a syntax error since this is an object literal, not a class definition). Are you reading the red line as the added line? The line as proposed in this PR is: 

```typescript
    controls: options.controls !== undefined ? options.controls : true,
```

When the incoming `options` object has no `controls` property defined, the value of the `controls` key in the resulting `videoJsOptions` becomes the default value that I specify, `true`. But when it _is_ defined in the incoming `options` object, _even if it is false_, it will not be overridden to be `true`, because that would sort of defeat the purpose of an option.

Reply:
Sorry, I misread the code and write to quickly (and my suggestions were totally wrong). I didn't saw that it was not parameters destructuring but type annotation (though it was obvious because you use `options.contol`). I agree with you it is outside the scope of this PR.

,